document/document.go

```diff
@@ -67,15 +67,78 @@ func (fb FieldBuffer) GetByField(field string) (Value, error) {
 	return Value{}, ErrFieldNotFound
 }
 
+// treatArrayValue update the value of array at the given index.
+func treatArrayValue(vlist Array, v Value, index int) (ValueBuffer, error) {
+	var buf ValueBuffer
+	err := vlist.Iterate(func(i int, va Value) error {
+		if index == i {
+			buf = buf.Append(v)
+			return nil
+		}
+		buf = buf.Append(va)
+		return nil
+	})
+
+	return buf, err
+}
+
+// SetDotNotation allow dot notation and replace value at the given index.
+func (fb *FieldBuffer) SetDotNotation(fname string, v Value) {
+	vPath := NewValuePath(fname)
+	index, strconvErr := strconv.Atoi(vPath[1])
+	for i, f := range fb.fields {
+		if f.Field != vPath[0] {
+			continue
+		}
+		switch f.Value.Type {
+		case DocumentValue:
+			var buf FieldBuffer
+			var b1 *FieldBuffer
+			// Cannot make field.integer on a document
+			if strconvErr == nil {
+				return
+			}
+
+			err := buf.Copy(f.Value.V.(Document))
+			if err != nil {
+				return
```

Comment:
In Go, if a function can fail the error must be returned so that the caller can make a decision about that. 
If `buf.Copy` fails, this entire function will fail, but as it is, this function doesn't indicate if the set worked or not.
You should simply return the error with `return err`

Reply:
Totally agree. I will correct it. Thank you